### PR TITLE
fix setDay examples for #1637

### PR DESCRIPTION
**Pull request by dcousens**
I think this is the intended functionality?

### CODE REVIEW

File: src/setDay/index.js
Comment (dcousens) [inline src/setDay/index.js:35]: @BoomDev it'd probably be a good idea to add this as a test too

Comment (oakify) [reply src/setDay/index.js:35]: @dcousens sounds great, will do.

One comment - and maybe this is being picky but please advise - the great majority of 'new Date()'
used in the code including the documentation is using this format:

new Date(2014, 8, 1)

Instead of your proposed:

new Date('2014-09-01')

...which I like better because it's more readable but wouldn't we want to be consistent with the rest?

Comment (oakify) [reply src/setDay/index.js:35]: wait... these test are already there!

@dcousens I think what you suggest is incorrect.

In setDay(date, **day**, [options]), my understanding was that 'day' is the day of the week of the new date (as in 0=Sun...), not the index of the day of the new date relative to the weekStartsOn.

In other words :
setDay(new Date('2014-09-01'), 0 /\*a Sunday\*/, { weekStartsOn: 1 })
=> **Sun** Sep 07 2014

And _not_:
setDay(new Date('2014-09-01'), 0 /\*index 0 of week starting on a Monday\*/, { weekStartsOn: 1 /\* Monday \*/})

**Review by oakify — CHANGES_REQUESTED** [review]
I would say:
 * @example
 * // Set week day to Sunday, with the default weekStartsOn of Sunday:
 * var result = setDay(new Date(2014, 8, 1) /* a Monday */, 0)
 * //=> Sun Aug 31 2014 00:00:00
 *
 * @example
 * // Set week day to Sunday, with a weekStartsOn of Monday:
 * var result = setDay(new Date(2014, 8, 1), 0, { weekStartsOn: 1 })
 * //=> Sun Sep 07 2014 00:00:00

File: src/setDay/index.js
Comment (dcousens) [reply src/setDay/index.js:35]: > ...which I like better because it's more readable but wouldn't we want to be consistent with the rest?

Indeed!  I couldn't help but question why the examples used that format for `new Date`.
I figured it either was OK or it wasn't.

Comment (dcousens) [reply src/setDay/index.js:35]: I think I need to review this function before I comment further,  the functionality isn't appearing straight forward to me.

Comment (oakify) [reply src/setDay/index.js:35]: @dcousens sure. The way I see it is, this is especially useful for locales where the week starts on Mondays (other weekStartsOn are edge cases I guess - as probably are the cases where day <0 or >6 , but OK).
So for example if you want to get the Sunday of a date's week, if you consider that that week starts on Mondays and your date is say a Tuesday, what you're looking for is the Sunday after that Tuesday, not the one before.
In contrast, what you suggested, taking the same example, was resulting on the Monday before that Tuesday. Which in fact you could get at by simply doing this:
setDay(new Date(2014, 9, 2 /* a Tuesday */), 1) 

Hope this helps.

Comment (dcousens) [reply src/setDay/index.js:35]: @BoomDev thank you for the explanation!   I understand now.

The `weekStartsOn` variable is for resolving the ambiguity of whether to use the Sunday from the before or after the given date.

Comment (oakify) [reply src/setDay/index.js:35]: @dcousens you're most welcome, happy it helped!

Comment (oakify) [reply src/setDay/index.js:35]: @dcousens what are the next steps ? Anything I can do to complete the process? Sorry for my ignorance, it's [shame!] my first contribution to open source...

Comment (dcousens) [reply src/setDay/index.js:35]: @BoomDev nothing to do here!  You PR is on the other repository and it is waiting for review from a maintainer.
I had simply hoped to contribute what I could here,  but,  as you thankfully pointed out,  my understanding was flawed.

:+1:

Comment (oakify) [reply src/setDay/index.js:35]: @dcousens thanks for your help! Cheers!

Comment (dcousens) [reply src/setDay/index.js:35]: PS.  Congratulations on contributing back to OSS!

Comment (oakify) [reply src/setDay/index.js:35]: > PS. Congratulations on contributing back to OSS!

Thanks! I'm glad I did. The code I wrote to fix the bug can probably be nicer but I though better contribute with working but ugly code than do nothing :-D

**Pull request closed without merging**
